test/unit/controllers/payment-reference/post-product-reference.controller.test.js

```diff
@@ -206,4 +206,48 @@ describe('product reference post controller', function () {
       expect($('.govuk-heading-m').text()).to.include('You can’t use any of the following characters')
     })
   })
+
+  describe('when reference entered is a potential PAN', function () {
```

Comment:
Maybe worth adding a quick test for when we post with a card number after clicking "Confirm and continue" to make sure it shows the amount page?

Reply:
Good point! added new test